hilti/src/compiler/parser/scanner.ll

```diff
@@ -33,6 +33,15 @@ static hilti::Meta toMeta(hilti::detail::parser::location l) {
     return hilti::Meta(hilti::Location(*l.begin.filename, l.begin.line, l.end.line));
 }
 
+static std::string expandEscapes(Driver* driver, std::string s, hilti::detail::parser::location l) {
+    try {
```

Comment:
Is there any reason to not perform more generic exception handling in the scope invoking the scanner/parser (which I think could be in HILTI's `parser/driver`? I am asking since the fix here just addresses #210, but not the more general issue of exceptions escaping from parsing.

Similarly in Spicy's scanner.

Reply:
Yeah, this way we can continue parsing and potentially catch & report more such errors before aborting. 

But you're right, the parser should have an outer catch on top of this.